lib/request-processor/utils/utils.go

```diff
@@ -228,33 +218,56 @@ func IsUserBlocked(userID string) bool {
 	return KeyExists(globals.CloudConfig.BlockedUserIds, userID)
 }
 
-func IsIpInBlocklist(ip string, ipBlocklist map[string]IpBlockList) (bool, []string) {
+type IpListMatch struct {
+	Key         string
+	Description string
+}
+
+func IsIpInList(ipList map[string]IpList, ip string) (int, []IpListMatch) {
+	if len(ipList) == 0 {
+		return NoConfig, []IpListMatch{}
+	}
+
 	ipAddress, err := netip.ParseAddr(ip)
 	if err != nil {
-		log.Infof("Invalid ip address: %s\n", ip)
-		return false, []string{}
+		return NoConfig, []IpListMatch{}
 	}
 
-	matchedDescriptions := []string{}
-	for _, ipBlocklist := range ipBlocklist {
-		if ipBlocklist.IpSet.Contains(ipAddress) {
-			matchedDescriptions = append(matchedDescriptions, ipBlocklist.Description)
+	matches := []IpListMatch{}
+	for listKey, list := range ipList {
+		if list.IpSet.Contains(ipAddress) {
+			matches = append(matches, IpListMatch{Key: listKey, Description: list.Description})
 		}
 	}
 
-	return len(matchedDescriptions) > 0, matchedDescriptions
+	if len(matches) == 0 {
+		return NotFound, matches
+	}
+
+	return Found, matches
+}
+
+func IsIpAllowed(ip string) (bool, []IpListMatch) {
+	globals.CloudConfigMutex.Lock()
+	defer globals.CloudConfigMutex.Unlock()
+	log.Debugf("Checking if IP %s is allowed", ip)
+	result, matches := IsIpInList(globals.CloudConfig.AllowedIps, ip)
+	// IP is allowed if it's found in the allowed lists or if the allowed lists are not configured
+	return result != NotFound, matches
```

Comment:
```suggestion
func IsIpAllowed(ip string) (bool, []IpListMatch) {
        // Always allow access from local IP addresses
	if helpers.IsPrivateIP(ip) {
		return true, matches
	}

	globals.CloudConfigMutex.Lock()
	defer globals.CloudConfigMutex.Unlock()
	log.Debugf("Checking if IP %s is allowed", ip)
	
	result, matches := IsIpInList(globals.CloudConfig.AllowedIps, ip)
	
	// If no allowed lists are configured, allow by default
	if result == NoConfig {
		return true, matches
	}
	
	// IP is allowed if it's found in the allowed lists
	return result == Found, matches
```

Reply:
We're not checking private IPs right now? https://github.com/AikidoSec/firewall-node/blob/main/library/agent/ServiceConfig.ts#L248-L251

Reply:
Maybe we can also drop the matches from the return type? We don't need 'em? Simplifies the function?